### PR TITLE
Access Protection + User Creation/Reading on Signup/Login

### LINKED ISSUE #3: Feature: User Login Page
## Epic
As a user, I want to login to the app

## User Stories
- As a user, I want to enter my username and password to sign in to my account
- As a user, I want to be redirected to the app based on my information 
- As a user, I want to be able to navigate to the register page if I don't have an account.

## List of Tasks
- [x] Upon submit of login information check if username / password is correct mapping (stored in a database), if not output error
- [x] Upon successful login, Redirect user to the app rendered with their information (correct profile page and swipe cards, stored in the database)

### LINKED ISSUE #13: Only allow access to a page if authenticated/logged-in (Back-end logic)
#8

### LINKED ISSUE #21: Create new User on Registration
When a new user makes an account, make a unique user object linked to their account.

### LINKED ISSUE #23: Access Protection + User Creation
## User Story
As a user, I want to make sure no one can access my profile except myself.

## Detailed Description
We need to add routing to the different pages that is protected by authentication. Users should only be able to see their own accounts. 

When a user signs up, a new User object should be created and linked to their account/email. The user should also be redirected to the Profile Page with the Edit Profile component open.
When a user signs in, their User object should be used to fill in details on the other pages. They should be redirected to the swiping page.
If a user tries to go to a page without having logged in, they should be redirected to the login page.

## Acceptance Criteria
- [x] Given successful SIGNUP, then create a new user object linked to a user's account. Redirect to profile page, with edit component open. #21 #25
- [x] Given successful LOGIN, then get corresponding user object. Redirect to swiping page (dependent on swiping page). #26 #28
- [x] Given attempt to access any page other than login without logging in, then redirect to login page. #13

### LINKED ISSUE #25: Given successful signup, redirect to profile page with edit component open

### LINKED ISSUE #26: Given successful LOGIN, get corresponding user object.

### LINKED ISSUE #28: Given successful login, redirect to swiping page

### PR DESCRIPTION
**Pull request by danieloh99**
Redirects to 'test' page on sign-up, #25 (just need to change final address)
Redirects to 'home' page on login, #28 (just need to change final address)
Creates new user in RTDB on sign up, #21. Currently has some dummy 'likes' and 'likedBy's.
Gets user object when logging in. Makes available to all other components via React Context. #26
Protects access so that only signed-in user can access pages. #13

This completes the story for #23. There are very minor changes like removing dummy content when creating a user, but this should be kept for now for testing purposes, especially while we do not have swiping features.

### CODE REVIEW

**Comment by jordanXYsoon1998** [comment]
Code looks good to me. Only suggestion I'd make would be to change the function signature of the `createNewUser()` async function. Rather than pass in the individual parameters, perhaps it could take in an object with key-value pairs to make it easier to use in other sections of the code?

File: src/services/firebase.js [hunk 1]
```diff
@@ -22,99 +21,37 @@ if (!firebase.apps.length) {
...
-        firebase.auth.EmailAuthProvider.PROVIDER_ID,
-        // firebase.auth.PhoneAuthProvider.PROVIDER_ID
-    ],
-    // Terms of service url.
-    tosUrl: 'https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstleyVEVO',
-    // Privacy policy url.
-    privacyPolicyUrl: 'https://www.youtube.com/watch?v=y6120QOlsfU&ab_channel=Darude'
+export const auth = firebase.auth;
+export const db = firebase.database();
+export const createNewUser = async (uid, name, photo, email, phone) => {
```
Comment (jordanXYsoon1998) [inline src/services/firebase.js]: Perhaps change to take in an options object with key-value pairs instead of multiple parameters?

**Review by jordanXYsoon1998 — COMMENTED** [review]
Code changes look good! Just one minor suggestion.

File: src/services/firebase.js (hunk 1 above)
Comment (danieloh99) [reply src/services/firebase.js]: Done

**Pull request merged**